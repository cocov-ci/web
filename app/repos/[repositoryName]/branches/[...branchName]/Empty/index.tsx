import { BoxSelect } from 'lucide-react'

import Alert from 'app/common/Alert'

import styles from './Empty.module.scss'

const Empty = () => {
  return (
    <div className={styles.empty}>
      <Alert
        description="This repository doesn't have a branch yet, or, if it has been recently added, its default branch is being processed. Try refreshing this page in a few seconds."
        icon={BoxSelect}
        title="Hmm. It's empty hereâ€¦"
      />
    </div>
  )
}

export default Empty
