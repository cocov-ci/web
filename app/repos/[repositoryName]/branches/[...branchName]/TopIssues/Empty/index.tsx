import { PartyPopper } from 'lucide-react'

import Alert from 'app/common/Alert'

import styles from './Empty.module.scss'

const Empty = () => {
  return (
    <div className={styles.empty}>
      <Alert
        className={styles.alert}
        description="Configured checks didn't find issues on this repository's default branch! Good job!"
        icon={PartyPopper}
        title="Yay! No Issues!"
      />
    </div>
  )
}

export default Empty
