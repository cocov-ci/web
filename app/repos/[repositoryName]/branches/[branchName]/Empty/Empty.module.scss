@import "styles/variables.module";

.empty {
  position: relative;

  .arrow {
    position: absolute;
    right: 80px;
    top: -10px;
  }
}
