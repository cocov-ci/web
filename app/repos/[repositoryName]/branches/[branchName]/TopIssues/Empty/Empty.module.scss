@import "styles/variables.module";

.empty {
  margin-bottom: 25px;

  .alert {
    > div {
      background-color: $colorGrayLight;
    }
  }
}
