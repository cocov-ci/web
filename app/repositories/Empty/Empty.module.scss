@import "styles/variables";

.empty {
  .arrow {
    position: absolute;
    right: 80px;
    top: -10px;
  }
}
