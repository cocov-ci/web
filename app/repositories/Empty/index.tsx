import { BoxSelect } from 'lucide-react'
import Image from 'next/image'

import Alert from 'app/common/Alert'

import styles from './Empty.module.scss'

const Empty = () => {
  return (
    <div className={styles.empty}>
      <Alert
        description="How about adding a new repository to get started?"
        icon={BoxSelect}
        title="Hmm. It's empty hereâ€¦"
      />

      <Image
        alt="arrow"
        className={styles.arrow}
        height="73"
        src="/icons/repositories/arrow.svg"
        width="187"
      />
    </div>
  )
}

export default Empty
