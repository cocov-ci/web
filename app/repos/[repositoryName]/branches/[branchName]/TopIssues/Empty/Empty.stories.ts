import type { Meta, StoryObj } from '@storybook/react'

import Empty from './index'

const meta: Meta<typeof Empty> = {
  title: 'Branches/[BranchName]/TopIssues/Empty',
  component: Empty,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof Empty>

export const Default: Story = {
  args: {},
}
