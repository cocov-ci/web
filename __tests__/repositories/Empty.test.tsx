import { render, screen } from '@testing-library/react'

import Empty from 'app/repositories/Empty'

import '@testing-library/jest-dom'

describe('Repositories/Empty', () => {
  it('renders Empty component', () => {
    const { container } = render(<Empty />)

    expect(
      screen.getByText('How about adding a new repository to get started?'),
    ).toBeVisible()
    expect(screen.getByText("Hmm. It's empty hereâ€¦")).toBeVisible()
    expect(container.querySelector('img')).toBeVisible()
  })

  it('renders Empty snapshots', () => {
    const { container } = render(<Empty />)
    expect(container).toMatchSnapshot()
  })
})
