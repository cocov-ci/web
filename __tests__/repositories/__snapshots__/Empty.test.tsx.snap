// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories/Empty renders Empty snapshots 1`] = `
<div>
  <div
    class="empty"
  >
    <div
      class="alert"
    >
      <div
        class="box"
      >
        <svg
          class="icon"
          fill="none"
          height="40"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 3a2 2 0 0 0-2 2"
          />
          <path
            d="M19 3a2 2 0 0 1 2 2"
          />
          <path
            d="M21 19a2 2 0 0 1-2 2"
          />
          <path
            d="M5 21a2 2 0 0 1-2-2"
          />
          <path
            d="M9 3h1"
          />
          <path
            d="M9 21h1"
          />
          <path
            d="M14 3h1"
          />
          <path
            d="M14 21h1"
          />
          <path
            d="M3 9v1"
          />
          <path
            d="M21 9v1"
          />
          <path
            d="M3 14v1"
          />
          <path
            d="M21 14v1"
          />
        </svg>
        <div
          class="content"
        >
          <h2>
            Hmm. It's empty hereâ€¦
          </h2>
          <h3>
            How about adding a new repository to get started?
          </h3>
        </div>
      </div>
    </div>
    <img
      alt="An arrow pointing to the Add Repository button"
      class="arrow"
      data-nimg="1"
      decoding="async"
      height="73"
      loading="lazy"
      src="/icons/repositories/arrow.svg"
      srcset="/icons/repositories/arrow.svg 1x, /icons/repositories/arrow.svg 2x"
      style="color: transparent;"
      width="187"
    />
  </div>
</div>
`;
